Sources/SATSCore/Components/WheelPickerView/WheelPickerView.swift

```diff
@@ -0,0 +1,131 @@
+import SwiftUI
+
+public struct WheelPickerView: View {
+    @Binding public var value: Int
+
+    private let circleThreshold: CGFloat = 300.0
+    private let knobPadding: CGFloat = 10.0
+
+    @State private var angleValue: CGFloat
+    @State private var numberOfRounds: CGFloat
+    @State private var previousAngleValue: CGFloat = 0.0
+
+    private let config: WheelConfig
+
+    public init(config: WheelConfig, value: Binding<Int>) {
+        self.config = config
+        _value = value
+
+        let floatValue = CGFloat(value.wrappedValue)
+        let remainder = floatValue.truncatingRemainder(dividingBy: config.totalValue)
+        let angle = 360 * remainder / config.totalValue
+        angleValue = angle
+        numberOfRounds = (floatValue / config.totalValue).rounded(.down)
+    }
+
+    public var body: some View {
+        ZStack {
+            // Background circle
+            Circle()
+                .fill(Color.backgroundSurfacePrimary)
+                .frame(width: config.radius * 2, height: config.radius * 2)
+                .scaleEffect(1.1)
+
+            // Ticks
+            Circle()
+                .stroke(
+                    Color.graphicalElements3,
+                    style: .init(
+                        lineWidth: config.sizeOfTicks,
+                        lineCap: .round,
+                        dash: dashes()
+                    )
+                )
+                .frame(width: config.radius * 2, height: config.radius * 2)
+
+            // Circle line
+            Circle()
+                .trim(from: 0.0, to: CGFloat(value) / config.totalValue)
+                .stroke(
+                    Color.graphicalElements5,
+                    style: .init(
+                        lineWidth: lineWidth(),
+                        lineCap: .round
+                    )
+                )
+                .frame(width: config.radius * 2, height: config.radius * 2)
+                .rotationEffect(.degrees(-90))
+                .pickerShadow
+
+            // Knob
+            Circle()
+                .fill(Color.graphicalElements5)
+                .frame(width: config.knobRadius * 2, height: config.knobRadius * 2)
+                .padding(knobPadding)
+                .offset(y: -config.radius)
+                .rotationEffect(Angle.degrees(Double(angleValue)))
+                .gesture(DragGesture(minimumDistance: 0.0)
+                    .onChanged({ value in
+                        change(location: value.location)
+                    })
+                )
+                .pickerShadow
```

Comment:
I don't know about SwiftUI, but if this was a Compose component, I would probably extract these elements into their own smaller components. Something like a

```swift
public struct WheelPickerView: View {
  // ...

  public var body: some View {
    ZStack {
      BackgroundCircle()
      Ticks()
      Line()
      Knob()
      
      if let label = config.label {
        Text("\(value) \(label)")
          .satsFont(.h1, weight: .emphasis)
          .foregroundColor(.onSurfacePrimary)
      }
    }
  }
}

private struct BackgroundCircle: View {
  // ...

  public var body: some View {
    Circle()
      .fill(Color.backgroundSurfacePrimary)
      .frame(width: config.radius * 2, height: config.radius * 2)
      .scaleEffect(1.1)
  }
}

private struct Ticks: View {
  // ...
}

private struct Line: View {
  // ...
}

private struct Knob: View {
  // ...
}
```

Does this incur an overhead in SwiftUI? Asking mostly out of curiosity. :sweat_smile:

Reply:
Personally I wouldn't go so far and do private structs, but could refactor to private props for the different parts instead of comments! Good feedback, though, Mr. Jobs. I'll update with that 👍